ir/src/main/scala/org/scalajs/core/ir/Printers.scala

```diff
@@ -500,19 +650,19 @@ object Printers {
           print("{}")
 
         case JSObjectConstr(fields) =>
-          print("{"); indent; println()
+          print('{'); indent; println()
           var rest = fields
           while (rest.nonEmpty) {
-            printOne(rest.head._1)
-            printOne(": ")
-            printOne(rest.head._2)
+            print(Objects.toString(rest.head._1))
```

Comment:
`rest.head._1` is a `PropertyName`, IIUC you need to dispatch it through the right print method, otherwise we have a (potential) degradation in source maps.